src/Bio/VCF/Parser/Parser.hs

```diff
@@ -1,22 +1,58 @@
 {-# LANGUAGE OverloadedStrings #-}
+
 module Bio.VCF.Parser.Parser where
 
 import Data.Attoparsec.ByteString
-import Data.Attoparsec.ByteString.Char8
+import qualified Data.Attoparsec.ByteString.Char8 as AC8
 import qualified Data.ByteString as B
-import Data.ByteString.Char8 (singleton)
+import qualified Data.ByteString.Char8 as BS8 (singleton, words)
+import Data.Word (Word8)
 import Bio.VCF.Internal.Types
 
+tabOrSpace :: Word8 -> Bool
+tabOrSpace c = c == 32 || c == 9
+
+isTab :: Word8 -> Bool
+isTab c = c == 9
+
+endOfLine :: Word8 -> Bool
+endOfLine c = c == 13 || c == 10
+
 parseMetaInformation :: Parser B.ByteString 
 parseMetaInformation = do
-  char '#'
-  char '#'
+  AC8.char '#'
+  AC8.char '#'
   rest <- takeByteString
   return rest
 
 parseFormatLine :: Parser B.ByteString
 parseFormatLine = do
-  char '#'
-  next <- fmap singleton (notChar '#')
+  AC8.char '#'
+  next <- fmap BS8.singleton (AC8.notChar '#')
   rest <- takeByteString
   return $ B.append next rest
+
+parsePatients :: Parser [B.ByteString]
```

Comment:
I'd define some `newtype`s for the resulting parsing results. That way you can tell the parser results from the input in the types of your functions.